coordinator/src/orderbook/db/orders.rs

```diff
@@ -306,6 +308,52 @@ pub fn get_by_trader_id_and_state(
         .optional()
 }
 
+/// Get all the filled matches for all the limit orders generated by `trader_id`.
+///
+/// This can be used to calculate the implicit position of the maker, assuming that all the filled
+/// matches were executed.
+pub fn get_all_limit_order_filled_matches(
+    conn: &mut PgConnection,
+    trader_id: PublicKey,
+) -> QueryResult<Vec<(Uuid, Decimal)>> {
+    let orders = orders::table
+        // We use `matches::match_order_id` so that we can verify that the corresponding app trader
+        // order is in `match_state` _`Filled`_. The maker's match remains in `Pending` (since the
+        // trade is not actually executed yet), which is not very informative.
+        .inner_join(matches::table.on(matches::match_order_id.eq(orders::trader_order_id)))
+        .filter(
+            orders::trader_id
+                .eq(trader_id.to_string())
+                // Looking for `Matched`, `Limit` orders only, corresponding to the maker.
+                .and(orders::order_type.eq(OrderType::Limit))
+                .and(orders::order_state.eq(OrderState::Matched))
+                // The corresponding app trader match is `Filled`.
+                .and(matches::match_state.eq(MatchState::Filled)),
+        )
+        .select((
+            orders::trader_order_id,
+            matches::quantity,
+            orders::direction,
+        ))
+        .load::<(Uuid, f32, Direction)>(conn)?;
+
+    let filled_matches = orders
+        .into_iter()
+        .map(|(order_id, quantity, direction_maker)| {
+            let quantity = Decimal::from_f32(quantity).expect("to fit into Decimal");
+
+            let quantity = match direction_maker {
+                Direction::Long => quantity,
+                Direction::Short => -quantity,
+            };
```

Comment:
I like this logic, maybe we should start using this accross the board.

Reply:
How do you think we should action your comment? I assume we are already mapping like this elsewhere.

Reply:
Oh, maybe we are already doing this :)